### PR TITLE
Disallow core boot if searching for third party drivers

### PR DESCRIPTION
**Pull request by mwhudson**
This disables core boot options if the user selected "search for third party drivers".

It also adapts the recent fix to not offer core boot options on non-UEFI systems to indicate that this is happening via a disallowed guided option.

### CODE REVIEW

**Comment by mwhudson** [comment]
Hmm this has a poor interaction with the autoinstall default for search_drivers -- basically, if any autoinstall file is passed at all, search_drivers defaults to True. Maybe we should think of a path towards changing that, or otherwise we need to be more cautious and not apply this as a default until we have checked to see if the user has asked for a core boot install or not (which is a bit of a layering violation, sigh)

File: subiquity/server/controllers/filesystem.py [hunk 1]
```diff
@@ -288,6 +289,11 @@ async def configured(self):
         self._configured = True
         if self._info is None:
             self.set_info_for_capability(GuidedCapability.DIRECT)
+        if (
+            self.app.base_model.source.search_drivers
+            is SEARCH_DRIVERS_AUTOINSTALL_DEFAULT
+        ):
+            self.app.base_model.source.search_drivers = not self.is_core_boot_classic()
```
Comment (dbungert) [inline subiquity/server/controllers/filesystem.py:296]: This ends up kind of a weird default.  I think we should note it in the autoinstall reference.
The reference will need update anyhow for core boot but I don't want to forget this part.

**Review by dbungert — APPROVED** [review]

File: subiquity/server/controllers/filesystem.py (hunk 1 above)
Comment (mwhudson) [reply subiquity/server/controllers/filesystem.py:296]: Yeah this behaviour is a bit strange. I've added a note.

Comment (dbungert) [reply subiquity/server/controllers/filesystem.py:296]: Thank you, I like the doc blurb

**Review by dbungert — APPROVED** [review]

File: subiquity/server/controllers/filesystem.py [hunk 2]
```diff
@@ -160,18 +193,11 @@ def capability_info_for_gap(
...
-                r.disallowed.append(
-                    GuidedDisallowedCapability(
-                        capability=capability,
-                        reason=GuidedDisallowedCapabilityReason.TOO_SMALL,
-                    )
-                )
-        else:
-            r.allowed = list(self.capability_info.allowed)
+        r = self.capability_info.copy()
+        if gap.size < install_min:
```
Comment (dbungert) [inline subiquity/server/controllers/filesystem.py:197]:
```suggestion
        if gap is None or gap.size < install_min:
```

Comment (dbungert) [reply subiquity/server/controllers/filesystem.py:197]: Minor conflict to resolve, something like this should do it

**Pull request merged**